src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/TypeBuilder.cs

```diff
@@ -473,37 +459,27 @@ public GCLayout(RuntimeTypeHandle rtth)
                 Debug.Assert(MethodTable != null);
 
                 _bitfield = null;
-                _isReferenceTypeGCLayout = false; // This field is only used for the LowLevelList<bool> path
                 _gcdesc = MethodTable->ContainsGCPointers ? (void**)MethodTable - 1 : null;
                 _size = (int)MethodTable->BaseSize;
             }
 
             /// <summary>
             /// Writes this layout to the given bitfield.
             /// </summary>
-            /// <param name="bitfield">The bitfield to write a layout to (may be null, at which
-            /// point it will be created and assigned).</param>
-            /// <param name="offset">The offset at which we need to write the bitfield.</param>
-            public void WriteToBitfield(LowLevelList<bool> bitfield, int offset)
+            /// <returns>The layout in bitfield.</returns>
+            public bool[] WriteToBitfield()
             {
-                ArgumentNullException.ThrowIfNull(bitfield);
-
                 if (IsNone)
-                    return;
+                    return TypeBuilderState.s_emptyLayout;
```

Comment:
I think this sentinel should be kelp private to `TypeBuilderState`. This check can deleted here - the one place that calls this method does this only when `IsNone` returns false.